stern: add page (#3517)
tldr-pages/tldr: pages/osx/stern.md

# Stern

> > Tail Logs form multiple pods and multiple containers on Kubernetes.

Author: oops..
thanks for reporting.
will correct it.

> > Reference: <https://github.com/wercker/stern>.

Author: okay..will change it.

- Tail logs of container "nginx" inside pod "nginx-pod" from namespace "production":

> 

Reviewer: Shall we replace the specific container name and pod name with a placeholder?

Author: done, thanks for suggesting.

> # Stern

Reviewer 2: S should be lower case.

Reviewer 3: Oh, my! I must have missed this.

> > Tail Logs from multiple pods and multiple containers on Kubernetes.

Reviewer 2: L should be lower case.

> More information: <https://github.com/wercker/stern>.

> - Tail logs from specific namespace:

Reviewer 3: Perhaps this will read better?
```suggestion
- Tail logs from a specific namespace:
```

> `stern <POD> --container <CONTAINER> --namespace <NAMESPACE>`

Reviewer 3: Perhaps we could change your token syntax here to match that of tldr pages?
```suggestion
`stern {{pod_name}} --container {{container}} --namespace {{namespace}}`
```

> - Tail logs using particular kubeconfig:

Reviewer 3: Perhaps this will read better?
```suggestion
- Tail logs using a particular configuration file:
```

> `stern <POD> --container <CONTAINER> --kubeconfig <KUBECONFIG_PATH>`

Reviewer 3: Perhaps we could change your token syntax here to match that of tldr pages?
```suggestion
`stern {{pod_name}} --container {{container}} --kubeconfig {{path/to/kubeconfig_file}}`
```

> - Tail logs since last 2 hours:

Reviewer 3: Perhaps this will read better?
```suggestion
- Display the last 2 hours worth of logs:
```

Author: i guess "Tail" is correct word because it displays last 2 hours worth of logs and tails for newer ones.
how about "Display the last 2 hours worth of logs and Tails for newer ones" ?

Reviewer 3: Ah, I see @senani-mayank. What about this then?
```suggestion
- Display the last 2 hours worth of logs, tailing for newer ones:
```

Author: that seems fine @sbrl .
thanks for suggestion.

> `stern <POD> --container <CONTAINER> --since 2h`

Reviewer 3: Perhaps this will read better?
```suggestion
`stern {{pod}} --container {{container}} --since {{2h}}`
```

> - Tail logs from pods having label "KEY1=VALUE1,KEY2=VALUE2" across all namespaces:

Reviewer 3: Perhaps this will read better?
```suggestion
- Tail logs from pods that have specific labels across all namespaces:
```

> `stern --container <CONTAINER> -l "KEY1=VALUE1,KEY2=VALUE2" --all-namespaces`

Reviewer 3: Perhaps this will read better?
```suggestion
`stern --container {{container}} -l "{{KEY1=VALUE1,KEY2=VALUE2}}" --all-namespaces`
```

- Tail logs from containers whose name starts with "XYZ":

> `stern <POD> --container "XYZ*"`

Reviewer 3: ```suggestion
`stern {{pod_name}} --container "{{XYZ*}}"`
```
